block/manager.go

```diff
@@ -575,6 +586,10 @@ func (m *Manager) produceBlock(ctx context.Context) error {
 }
 
 func (m *Manager) submitNextBatch(ctx context.Context) {
+	defer func() {
```

Comment:
@omritoptix this allows the submit process to recover and try again in case of failure